pkgs/development/libraries/qt-5/5.7/qtbase/default.nix

```diff
@@ -91,106 +77,132 @@ stdenv.mkDerivation {
+    -shared
     -largefile
     -accessibility
-    -rpath
-    -optimized-qmake
-    -strip
-    -reduce-relocations
-    -system-proxies
+
+    -no-sql-db2
+    -no-sql-ibase
+    ${lib.optionalString (null != mysql) "-plugin-sql-mysql"}
+    ${lib.optionalString (null == mysql) "-no-sql-mysql"}
+    -no-sql-oci
+    -no-sql-odbc
+    -no-sql-sqlite2
+    ${lib.optionalString (null != postgresql) "-plugin-sql-psql"}
+    ${lib.optionalString (null == postgresql) "-no-sql-psql"}
+    -no-sql-tds
+
+    -system-sqlite
+    -qml-debug
     -pkg-config
 
+    -system-zlib
+    ${lib.optionalString stdenv.isDarwin "-I${zlib.out}/include/ -L${zlib.out}/lib"}
+    -no-mtdev
+    -system-libpng
+    ${lib.optionalString stdenv.isDarwin "-I${libpng.dev}/include/ -L${libpng.out}/lib"}
+    -system-libjpeg
+    ${lib.optionalString stdenv.isDarwin "-I${libjpeg.dev}/include/ -L${libjpeg.out}/lib"}
+    -system-doubleconversion
+    ${lib.optionalString stdenv.isDarwin "-I${double_conversion.out}/include/ -L${double_conversion.out}/lib"}
+    ${lib.optionalString stdenv.isLinux "-system-freetype"}
+    ${lib.optionalString stdenv.isDarwin "-no-freetype"}
+    -system-harfbuzz
+    ${lib.optionalString stdenv.isDarwin "-I${harfbuzz.dev}/include/ -I${harfbuzz.out}/include/"}
+    ${lib.optionalString stdenv.isDarwin "-L${harfbuzz.out}/lib"}
+    ${lib.optionalString stdenv.isLinux "-openssl-linked"}
+    ${lib.optionalString stdenv.isDarwin "-no-openssl"}
+    -no-libproxy
+    -system-pcre
+    ${lib.optionalString stdenv.isDarwin "-I${pcre16.dev}/include/ -L${pcre16.out}/lib"}
+    -system-xcb
+    ${lib.optionalString stdenv.isDarwin "-no-xkbcommon-x11"}
+    ${lib.optionalString stdenv.isDarwin "-no-xkbcommon-evdev"}
+    -${lib.optionalString stdenv.isDarwin "no-"}glib
+    ${lib.optionalString stdenv.isDarwin "-no-pulseaudio"}
+    ${lib.optionalString stdenv.isDarwin "-no-alsa"}
+    -${lib.optionalString stdenv.isDarwin "no-"}gtk
+
+    -make libs
+    -make tools
+    -${lib.optionalString (buildExamples == false) "no"}make examples
+    -${lib.optionalString (buildTests == false) "no"}make tests
+
     -gui
     -widgets
-    -opengl desktop
-    -qml-debug
+    -rpath
+    -verbose
+    -silent
     -nis
+    -${lib.optionalString (cups == null) "no-"}cups
     -iconv
+    ${lib.optionalString stdenv.isDarwin "-no-evdev"}
+    -no-tslib
     -icu
+    ${lib.optionalString stdenv.isDarwin "-I${icu.dev}/include/ -L${icu.out}/lib"}
+    -${lib.optionalString stdenv.isDarwin "no-"}fontconfig
+    -strip
     -pch
-    -glib
-    -xcb
-    -qpa xcb
-    -${lib.optionalString (cups == null) "no-"}cups
-
+    -dbus-linked
+    ${lib.optionalString stdenv.isLinux "-reduce-relocations"}
+    ${lib.optionalString stdenv.isDarwin "-no-use-gold-linker"}
+    -${lib.optionalString stdenv.isDarwin "no-"}xcb
     -no-eglfs
+    -no-kms
+    -no-gbm
     -no-directfb
     -no-linuxfb
-    -no-kms
-
-    ${lib.optionalString (!system-x86_64) "-no-sse2"}
```

Comment:
IIUC, in the absence of these flags, Qt will try to detect these features based on the host CPU. That makes builds non-deterministic and it can actually result in binary incompatibility!